src/InterpreterGenerator.h

```diff
@@ -202,7 +202,7 @@ class NodeGenerator : public RamVisitor<std::unique_ptr<InterpreterNode>> {
 
     NodePtr visitIndexScan(const RamIndexScan& scan) override {
         NodePtrVec children;
-        for (const auto& value : scan.getRangePattern()) {
+        for (const auto& value : scan.getRangePattern().second) {
```

Comment:
We need to fix this so that upper and lower bounds are stored in an interpreter node.